packages/tools/client-debugger/client-debugger-view/src/Debugger.tsx

```diff
@@ -32,48 +39,68 @@ export interface FluidClientDebuggerProps {
  * @remarks If no debugger has been initialized, will display a note to the user and a refresh button to search again.
  */
 export function FluidClientDebugger(props: FluidClientDebuggerProps): React.ReactElement {
-    /**
-     * This function will retrieve all client debuggers and return the first one.
-     * TODO: update it to check container id after support multi-containers app.
-     */
     function getFirstDebugger(): IFluidClientDebugger | undefined {
-        const clientDebuggers = getFluidClientDebuggers();
-        return clientDebuggers.length === 0 ? undefined : clientDebuggers[0];
+        const debuggers = getFluidClientDebuggers();
+        return debuggers.length === 0 ? undefined : debuggers[0];
     }
 
+    const [clientDebuggers, setClientDebuggers] = React.useState<IFluidClientDebugger[]>(getFluidClientDebuggers());
+
+    const [selectedContainerId, setSelectedContainerId] = React.useState<string>(
+        getFirstDebugger()?.containerId ?? "",
+    );
+
     const [clientDebugger, setClientDebugger] = React.useState<IFluidClientDebugger | undefined>(
-        getFirstDebugger(),
+        selectedContainerId === undefined ? undefined : getFluidClientDebugger(selectedContainerId),
     );
 
     const [isContainerDisposed, setIsContainerDisposed] = React.useState<boolean>(
         clientDebugger?.disposed ?? false,
     );
 
+    const debuggerRegistry: DebuggerRegistry = getDebuggerRegistry();
```

Comment:
We're already getting the full list of debuggers above. We probably want to move this higher up, and get the list of debuggers from this directly.

Reply:
I think monitor client debugger's change in here is reasonable. The higher up is `RenderClientDebugger`. Or should I have a debugger controller tsx to be the higher up call?

Reply:
I'm sorry, I completely missed your response here earlier.

I'm suggesting that since we're accessing the full list of debuggers on line 47 via `getFluidClientDebuggers` (which internally uses the registry), it might be nice to move this line above that, and replace the call to `getFluidClientDebuggers` with a call to `debuggerRegistry.getRegisteredDebuggers`.

This isn't a huge deal, but it would remove a lookup call on the window object, and I think it makes the logical flow of the function a little nicer.

Reply:
Ah, I see! Fixed it!